pipeline/src/TruthSimulationConfig.jl

```diff
@@ -0,0 +1,74 @@
+"""
+A configuration struct for truth simulation to compare to inference.
+
+NB: This config assumes Gamma distributed generation interval, specified by
+mean and standard deviation.
+"""
+@kwdef struct TruthSimulationConfig{T <: Real, F <: Function}
+    "True time-varying process. Default this is the time-varing reproductive
+    number"
+    truth_process::Vector{T}
+    "True generation interval distribution mean."
+    gi_mean::T
+    "True generation interval distribution std."
+    gi_std::T
+    "Infection-generating model type. Default is `:Renewal`."
+    igp::UnionAll = Renewal
+    "Maximum next generation interval when discretized. Default is 21 days."
+    D_gen::T = 60.0
+    "Transformation function"
+    transformation::F = exp
+    "Delay distribution: Default is Gamma(4, 5/4)."
+    delay_distribution::Distribution = Gamma(4, 5 / 4)
+    "Maximum delay when discretized. Default is 15 days."
+    D_obs::T = 15.0
+    "Prior for log initial infections. Default is Normal(4.6, 1e-5)."
+    log_I0_prior::Distribution = Normal(log(100.0), 1e-5)
+    "Prior for negative binomial cluster factor. Default is HalfNormal(0.1)."
+    cluster_factor_prior::Distribution = HalfNormal(0.1)
+end
+
+"""
+    simulate_or_infer(config::TruthSimulationConfig)
+
+Simulates or infers the truth process and observations based on the given configuration.
+
+# Arguments
+- `config::TruthSimulationConfig`: The configuration object containing the parameters for the simulation.
+
+# Returns
+A dictionary containing the sampled infections and observations, along with other relevant information.
+
+"""
+function simulate_or_infer(config::TruthSimulationConfig)
```

Comment:
Can you clarify why we want this to be simulate or infer vs having simulate and infer functions

Reply:
Only one reason: makes the overall code pattern even more similar on the next PR. But happy to change this. I was clearly being influenced by reading about `DrWarson.produce_or_load`.

Reply:
Ha fair enough. Shall we pin it as an issue to discuss once we have the next PR?